packages/core/src/utils/summarizer.ts

```
@@ -129,3 +208,256 @@
+  const breakPoints = ['\n\n', '\n}', '\n)', ';'];
+
+  for (const breakStr of breakPoints) {
+    const lastBreak = content.lastIndexOf(breakStr, truncatePoint);
+    if (lastBreak > truncatePoint * 0.8) {
+      breakPoint = lastBreak + breakStr.length;
+      break;
+    }
+  }
+
+  return (
+    content.substring(0, breakPoint) + '\n\n... (truncated for token budget)'
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)